### PR TITLE
feat: MavenLauncher Warning when a provided dependency is not loaded 

### LINKED ISSUE #1621: feat: SpoonMavenLauncher display warnings when presence of provided dependencies in pom.xml
When a pom.xml contains provided dependencies, they still might be mandatory to build the model: it would be useful to the user to see that they were missing dependencies (because provided) on the form of a console warning.

### PR DESCRIPTION
**Pull request by surli**
Fix #1621

### CODE REVIEW

File: src/main/java/spoon/MavenLauncher.java [hunk 1]
```diff
@@ -281,10 +282,10 @@ private String extractVariable(String value) {
 				}
 				// ignore not transitive dependencies
 				if (isLib && ("test".equals(dependency.getScope()) || "provided".equals(dependency.getScope()))) {
+					LOGGER.log(Level.WARN, "Dependency ignored (scope: provided or test): " + dependency.toString());
 					continue;
 				}
 				String fileName = dependency.getArtifactId() + "-" + version;
-				// TODO: Check the scope of the dependency (local dependency is not handled)
```
Comment (tdurieux) [inline src/main/java/spoon/MavenLauncher.java]: this TODO is still not handled

**Comment by tdurieux** [comment]
This  TODO is still not handled "Check the scope of the dependency (local dependency is not handled)"

**Comment by surli** [comment]
> this TODO is still not handled "Check the scope of the dependency

Isn't it the job of this condition? Else what do you mean by handle them?

**Comment by tdurieux** [comment]
Now I was lazy I did not handle System dependencies

**Pull request merged**